### DISCUSSION TITLE
Backup on Windows, Restore on Linux fails due to backslashes in filenames stored in sqlite

### DISCUSSION THREAD

**Discussion by stuartvsmith** [GAM-team/got-your-back#418] [Ideas]
TL,DR: Does the sqlite db file created for a GYB backup contain platform specific directory separators that make cross platform restore impossible?  If so, could this be improved, or could a note be added to the wiki to help people who encounter it?

Several years ago, I created a backup of a key Workspace user using GYB, before ultimately deleting the account.  At the time I used GYB installed on a Windows laptop.

Time passed, and I was issued a new Windows laptop, but rather than reinstalling GYB, I switched to using GYB in an EC2 instance.  I kept doing backups & restores as needed without issue.

Yesterday I was asked if it was possible to recover an email from the backup I had created on Windows.  I transferred the file to the EC2 instance where I use GYB and started the restore as usual, only to be confronted with a hundreds of error messages like this, and GYB exiting:

WARNING! file /home/USERNAME/GYB-GMail-Backup-FIRSTNAME.LASTNAME@DOMAIN.com/2021\1\29\1774fd471fb45684.eml does not exist for message 1688
  this message will be skipped.

After noticing the mix of backslashes & forward slashes I reinstalled GYB on Windows and tried the restore again and everything went fine.

Have I got this right is there something else going on or perhaps a workaround I missed?
Could GYB one day either store paths in a platform agnostic way?
Could a note be added to the wiki noting that cross platform restores won't work?

Thanks!

**Comment by arpanghosh8453** [comment]
Here, you can use this python script to solve the issue:

```
import sqlite3

con = sqlite3.connect("path\to\your\msg-db.sqlite")
cursor = con.cursor()
cursor.execute("UPDATE messages SET message_filename = REPLACE(message_filename, '\\', '/')")
con.commit()
con.close()
```
